#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.0016704714876199951 0.8917006284230288 0.4526225787507562 3.121555088208343
  position 0.05248484066957293 1.7361647256182922 -1.4272954401556441
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 1.25 1.25
}
Robot {
  translation 0.25 0.038 -0.5
  rotation 0 1 0 4.692820414042842e-06
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0.045 -0.0176 0
      }
      device [
        RotationalMotor {
          name "motor roda esquerda"
        }
        PositionSensor {
          name "sensor roda esquerda"
        }
      ]
      endPoint Solid {
        translation 0.045004127923461694 -0.01760095436205577 1.3234967960943285e-06
        rotation -0.999999647772517 -0.0006327988091038336 -0.0005513805483559795 5.758821796483726
        children [
          Transform {
            rotation 0 0 1 1.5707996938995747
            children [
              DEF shape_roda Shape {
                appearance PBRAppearance {
                  baseColor 0.8 0.15 0.3
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "roda direita"
        boundingObject Transform {
          rotation 0 0 1 1.5707996938995747
          children [
            USE shape_roda
          ]
        }
        physics DEF fisica_roda Physics {
          density -1
          mass 0.005
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor -0.045 -0.0176 0
      }
      device [
        RotationalMotor {
          name "motor roda direita"
        }
        PositionSensor {
          name "sensor roda direita"
        }
      ]
      endPoint Solid {
        translation -0.045 -0.0176 1.3235e-06
        rotation 0 1 0 1.0150305250839343e-06
        children [
          Transform {
            rotation 0 0 1 1.5707996938995747
            children [
              USE shape_roda
            ]
          }
        ]
        name "roda esquerda"
        boundingObject Transform {
          rotation 0 0 1 1.5707996938995747
          children [
            USE shape_roda
          ]
        }
        physics USE fisica_roda
      }
    }
    DEF rus_shape Shape {
      appearance PBRAppearance {
        baseColor 0.58 0.4 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
    DistanceSensor {
      translation 0.00130537 0.01 -0.0379144
      rotation 0 1 0 1.5708
      children [
        Transform {
          rotation 0 0 1 -1.5708003061004252
          children [
            DEF DIST_SENSOR_SHAPE_1 Shape {
              appearance DEF DIST_SENSOR_APAR PBRAppearance {
                baseColor 0.9 0.9 0
                roughness 1
                metalness 0
              }
              geometry DEF DIST_SENSOR_SHAPE Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ir0"
      lookupTable [
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
    }
    DistanceSensor {
      translation 0 0.01 0.038
      rotation 0 -1 0 1.5708
      name "ir1"
      lookupTable [
        0.03 1024 0
        0.2 0 0
      ]
      numberOfRays 2
    }
    DistanceSensor {
      translation 0.04 0.01 -0.0019996
      rotation 0 1 0 -5.307179586466759e-06
      name "ir2"
      lookupTable [
        0.03 1024 0
        0.2 0 0
      ]
      numberOfRays 2
    }
    DistanceSensor {
      translation -0.04 0.01 -0.00199984
      rotation 0 1 0 3.14159
      name "ir3"
      lookupTable [
        0.03 1024 0
        0.2 0 0
      ]
      numberOfRays 2
    }
  ]
  name "rus"
  boundingObject DEF rus_shape Shape {
    appearance PBRAppearance {
      baseColor 0.58 0.4 1
      roughness 1
      metalness 0
    }
    geometry Box {
      size 0.075 0.075 0.075
    }
  }
  physics Physics {
    density -1
    mass 0.2
  }
  controller "rus_controller"
  synchronization FALSE
}
Solid {
  translation -0.38 0.0476 -0.5
  rotation 0 1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  boundingObject USE parede
}
Solid {
  translation -0.5 0.0476 -0.12
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(1)"
  boundingObject USE parede
}
Solid {
  translation -0.25 0.0476 0.130001
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(7)"
  boundingObject USE parede
}
Solid {
  translation -0.38 0.0476 0.5
  rotation 0 1 0 1.5708
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
                baseColor 0 1 0
                metalness 0.5
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(9)"
  boundingObject USE parede
}
Solid {
  translation 0.25 0.0476 0.38
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(10)"
  boundingObject USE parede
}
Solid {
  translation -6.36856e-07 0.0476 0.13
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(11)"
  boundingObject USE parede
}
Solid {
  translation 6.89943e-07 0.0476 0.38
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(3)"
  boundingObject USE parede
}
Solid {
  translation 0.25 0.0476 -0.130001
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(14)"
  boundingObject USE parede
}
Solid {
  translation 0.25 0.0476 -0.370001
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                metalness 0.5
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(5)"
  boundingObject USE parede
}
Solid {
  translation 0.38 0.0476 0.25
  rotation 0 1 0 1.5708
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(12)"
  boundingObject USE parede
}
Solid {
  translation 0.37 0.0476 -0.5
  rotation 0 1 0 1.5708
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
                baseColor 1 0 0.0156863
                metalness 0.5
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(13)"
  boundingObject USE parede
}
Solid {
  translation -0.12 0.0476 -0.25
  rotation 0 1 0 1.5708
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(8)"
  boundingObject USE parede
}
Solid {
  translation -0.5 0.0476 0.13
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(2)"
  boundingObject USE parede
}
Solid {
  translation -0.12 0.0476 -0.5
  rotation 0 1 0 -1.5707953071795862
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(15)"
  boundingObject USE parede
}
Solid {
  translation 0.0100025 0.0476 -0.13
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(6)"
  boundingObject USE parede
}
Solid {
  translation -0.25 0.0476 0.38
  rotation 0 -1 0 -5.307179586466759e-06
  children [
    Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
                baseColor 0 1 0
                metalness 0.5
              }
              geometry Box {
                size 0.25 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  name "solid(4)"
  boundingObject USE parede
}
